TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/android/LoadOccupancyGridDialog.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2017 Intermodalics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.intermodalics.tango_ros_streamer.android;
+
+import android.app.DialogFragment;
+import android.content.Context;
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import eu.intermodalics.tango_ros_streamer.R;
+
+
+/**
+ * Displays dialog requesting name of the map before saving.
+ */
+public class LoadOccupancyGridDialog extends DialogFragment implements View.OnClickListener {
+    private static final String TAG = LoadOccupancyGridDialog.class.getSimpleName();
+
+    Button mOkButton;
+    EditText mNameEditText;
+    CallbackListener mCallbackListener;
+
+    public interface CallbackListener {
+        void onClickOkLoadOccupancyGridDialog(String name);
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        mCallbackListener = (CallbackListener) context;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflator, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View dialogView = inflator.inflate(R.layout.dialog_load_occupancy_grid, null);
+        getDialog().setTitle(R.string.load_occupancy_grid_dialog_title);
+        mOkButton = (Button) dialogView.findViewById(R.id.load_occupancy_grid_ok);
+        mOkButton.setOnClickListener(this);
+        dialogView.findViewById(R.id.load_occupancy_grid_cancel).setOnClickListener(this);
+        mNameEditText = (EditText) dialogView.findViewById(R.id.load_occupancy_grid_name);
+        mNameEditText.addTextChangedListener(new TextWatcher() {
```

Comment:
Can you replace the name edit text field by a list with occupancy grids that are available to load?